tilavarauspalvelu/integrations/email/template_context/common.py

```diff
@@ -225,11 +229,28 @@ def get_varaamo_ext_link(*, language: Lang) -> str:
     return url_base
 
 
-def get_my_applications_ext_link(*, language: Lang) -> str:
-    url_base = settings.EMAIL_VARAAMO_EXT_LINK.removesuffix("/")
+def get_my_applications_ext_link(
+    *,
+    language: Lang,
+    application_section: ApplicationSection | None = None,
+) -> str:
+    """
+    Return the link to the 'My applications' page:
+    e.g. https://varaamo.hel.fi/applications/{application_id}/view?tab=reservations&section={application_section_id}
+    """
+    url = settings.EMAIL_VARAAMO_EXT_LINK.removesuffix("/")
+
     if language != "fi":
-        return f"{url_base}/{language}/applications"
-    return f"{url_base}/applications"
+        url = f"{url}/{language}"
+    url = f"{url}/applications"
+
+    if application_section:
+        application_id = getattr(application_section, "application_id", None)
+        application_section_id = getattr(application_section, "id", None)
+
+        url = f"{url}/{application_id}/view?tab=reservations&section={application_section_id}"
+
+    return url
```

Comment:
I would create a separate function to create the section link. Then I would give that function more specific info instead of the whole `ApplicationSection`, e.g. just the `application_id` and `application_section_id`, which would allow us to generate the same link using test data.